docs/content/sips/008-using-oci-registries.md

```diff
@@ -0,0 +1,298 @@
+      "digest": "sha256:55c29ad4b0ad0c6bd8ec1ffc8f04e63342e5901280037ef706b1b114475d3cbb",
+      "size": 2147122
+    },
+    {
+      "mediaType": "application/vnd.wasm.content.layer.v1+data",
+      "digest": "sha256:a4699e4f9ef3f4922f38f0d017aa26438908f38caf020a739e0ee27fe796eb02",
+      "size": 178
+    }
+  ]
+}
+```
+
+Note: the media types used are not final and _can_ change based on the community
+standardization for Wasm modules in OCI registries.
+
+The manifest contains layers for the Wasm module for the component and for the
+one static asset referenced by the component. For each additional component, the
+Wasm module and static assets would be individual layers in the manifest above.
+
+Let's explore the OCI configuration object referenced in the manifest -
+`config.json` - it is a locked application manifest that Spin can use to run the
+application from the local cache:
+
+```json
+{
+  "spin_lock_version": 0,
+  "metadata": {
+    "description": "",
+    "name": "github-stars-webhook",
+    "trigger": {
+      "base": "/",
+      "type": "http"
+    },
+    "version": "0.1.0"
+  },
+  "triggers": [
+    {
+      "id": "trigger--github-star-webhook",
+      "trigger_type": "http",
+      "trigger_config": {
+        "component": "github-star-webhook",
+        "executor": null,
+        "route": "/..."
+      }
+    }
+  ],
+  "components": [
+    {
+      "id": "github-star-webhook",
+      "metadata": {
+        "allowed_http_hosts": ["https://hooks.slack.com"]
+      },
+      "source": {
+        "content_type": "application/wasm",
+        "digest": "sha256:55c29ad4b0ad0c6bd8ec1ffc8f04e63342e5901280037ef706b1b114475d3cbb"
+      },
+      "files": [
+        {
+          "digest": "sha256:a4699e4f9ef3f4922f38f0d017aa26438908f38caf020a739e0ee27fe796eb02",
+          "path": "my-file.json"
+        }
+      ]
+    }
+  ]
+}
+```
+
+This is all the information required for Spin to be able to push, pull, then run
+an application from an OCI registry.
+
+### Impact to `spin deploy`
+
+The first step in `spin deploy` is pushing the application to the registry. In
+line with the gradual approach to this change, `spin deploy` will also change to
+publish the application to an OCI registry instead, and by the time this change
+will be default, Fermyon Platform and Fermyon Cloud will support this change.
+
+The goal is to have at least one minor release of Spin with `spin oci` before
+updating the `spin deploy` behavior to expect an OCI registry.
+
+### Implementation status
+
+The `spin oci push`, `spin oci pull`, and `spin oci run` commands are currently
+implemented in this prototype. The implementation uses the
+[`oci-distribution`](https://github.com/krustlet/oci-distribution) crate from
+Krustlet to interact with a container registry (and currently uses a fork that
+should be patched upstream).
+
+The internals of the loaders need additional work before being merged, and the
```

Comment:
Are you amenable to merging the SIP separately from the code?  Just thinking it will be easier to collaborate on the implementation (and reconcile with parallel changes) in PRs to main than to a branch in your fork.

Reply:
Absolutely.
With that in mind, I would propose we go through a full round of reviews and updates here, and when we are happy with the state of the SIP, break it out into its own PR.